source/includes/tutorial/webdev_mapbox_quickstart/_quickstart_webmap.md

```diff
@@ -0,0 +1,492 @@
+
+The API response obtained earlier contains a reference to a tile layer and a URL, which for your own benefit should be found within the metadata by skimming the contents.
+If you are unable to find it, look within the following hierarchy address:
+
+<div class="center-column"></div>
+```
+.data.attributes.layer[0].attributes.layerConfig.source.tiles[0]
+```
+
+Datasets and Layers accessed through the API are not guaranteed to have the same structuring of their `layerConfig` object, which means there may be a need to investigate each dataset in a more comprehensive way than is being demonstrated here.
+Since this is a tutorial, some of this logic is going to be hardcoded for now.
+
+For the given dataset, you will see a tile layer URL that looks like:
+
+<div class="center-column"></div>
+```
+https://tiles.globalforestwatch.org/umd_tree_cover_loss/v1.7/tcd_{thresh}/{z}/{x}/{y}.png
+```
+
+This URL is close to what is needed by Mapbox, but there is a templated component `{thresh}` which will not be suitable for Mapbox to consume.
+In this case, URL is templated beyond what is expected by the application and may be troublesome.
+By browsing the `layer` object in the API response it is possible to find some additional attributes for how the layer parameters are configured.
+For this specific case, there is a parameterization array at the address
+
+<div class="center-column"></div>
+```
+.data.attributes.layer[0].attributes.layerConfig.params_config
+```
+
+That parameterization array looks like:
+
+<div class="center-column"></div>
+```
+...
+"params_config": [
+  {
+    "default": 30,
+    "key": "thresh",
+    "required": true
+  }
+],
+...
+```
+
+By using this parameterization data, the URL can be transformed into compliance with Mapbox GL.
+
+Update the `<body>` section of `index.html` with the following javascript, a description is located after this code block.
+
+<div class="center-column"></div>
+```html
+  <!-- JS embedded in body tag  -->
+  <script>
+    // ...
+    // callApiDatasetMetadata = async function ....
+    // ...
+
+    // get the Mapbox-ready raster tile URL template (example.com/{x}/{y}/{z})
+    // takes as a single parameter the API response data
+    getTileLayerUrlForTreeCoverLoss = function(obj) {
+      // drill down to get a useful object
+      var layerConfig = obj['data']['attributes']['layer'][0]['attributes']['layerConfig'];
+      // get the full templated URL
+      var urlTemplate = layerConfig['source']['tiles'][0];
+      // get the URL template parameters
+      var defaultParams = layerConfig['params_config'];
+
+      // define reducing function that iteratively substitutes parameters
+      // takes as two parameters:
+      //   (0) the accumulated string of the URL with template components substituted
+      //   (1) the current object in the parameters array
+      urlTemplateReducer = function(accumulated, value) {
```

Comment:
I'm not sure how great I feel about using a reducer in an intro tutorial. It's not a super familiar pattern for a lot of people and in this case we're just substituting the one parameter.

Reply:
To be clear, I am suggesting that we just do a for loop instead.

Reply:
Handled in JS refactor of bd66b37